src/legacy/server/saved_objects/routes/bulk_update.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Hapi from 'hapi';
+import Joi from 'joi';
+import {
+  SavedObjectAttributes,
+  SavedObjectsUpdateOptions,
+  SavedObjectsClient,
+} from 'src/core/server';
+import { Prerequisites } from './types';
+
+interface SavedObject {
+  type: string;
+  id: string;
+  attributes: SavedObjectAttributes;
+  options: SavedObjectsUpdateOptions;
+}
+
+interface BulkUpdateRequest extends Hapi.Request {
+  pre: {
+    savedObjectsClient: SavedObjectsClient;
+  };
+  payload: SavedObject[];
+}
+
+export const createBulkUpdateRoute = (prereqs: Prerequisites) => {
+  return {
+    path: '/api/saved_objects/_bulk_update',
+    method: 'PUT',
+    config: {
+      pre: [prereqs.getSavedObjectsClient],
+      validate: {
+        payload: Joi.array().items(
+          Joi.object({
+            type: Joi.string().required(),
+            id: Joi.string().required(),
+            attributes: Joi.object().required(),
+            version: Joi.string(),
+            namespace: Joi.string(),
```

Comment:
Let's remove `namespace` as an expected property. This isn't a field that we want or expect consumers to provide or be concerned about.

Reply:
It is now under the Options argument instead, which is at method level rather than entity level.

Does that make more sense?

Reply:
Making `namespace` part of the `options` argument from the perspective of the `SavedObjectsClient` makes sense, but from the API perspective, `namespace` should be removed entirely. The Spaces plugin is responsible for providing the `namespace` attribute to the SavedObjectsClient whenever necessary, via its SavedObjectsClient wrapper

Reply:
I see, ok, I'll remove it then. 👍

Reply:
I've removed namespace (hence all options, as that was the only one) for spaces.
I've also removed it from the SavedObjectsClient as it communicates via the API and so, there isn't any point having the option there if it can't be used via the API, right?

Reply:
Ah, we still need it on the SavedObjectsClient -- the instance the API uses is "wrapped" by the spaces plugin, and the spaces plugin is what effectively passes in the `namespace` attribute down to the repository:

API --> Spaces Saved Objects Client --> Base (OSS) Saved Objects Client --> Saved Objects Repository.

The fact that the SOC is wrapped is transparent to all consumers. You aren't aware that you're dealing with a wrapped instance. If the `namespace` option is removed from the client, then the spaces wrapper can't effectively write the objects into the correct space. In fact, it will only ever work for the default space in this case, as the default space is denoted by the absence of a namespace

Reply:
Hmm interesting, shouldn't have the integration tests failed then when I removed them?
because they passed locally 🤔

Reply:
Discussed synchronously and looks like it might be right, @legrego will review.